stdlib/public/core/StringBridge.swift

```diff
@@ -737,6 +737,22 @@ internal func _SwiftCreateBridgedString_DoNotCall(
   return Unmanaged<AnyObject>.passRetained(str._bridgeToObjectiveCImpl())
 }
 
+extension String {
+  @_spi(Foundation)
+  public init<Encoding: Unicode.Encoding>(_immortalCocoaString: AnyObject, buffer: UnsafeBufferPointer<UInt8>, encoding: Encoding.Type) {
```

Comment:
The idea here is that Foundation would identify constant ASCII NSStrings (it can do this cheaply) and call this instead of the bridging funnel point